[bug] Native Anthropic reviewer path fails with "Unable to connect" under v1 action

## User Story

As a repo maintainer, I want the reviewer to optionally run on native Anthropic Sonnet (not just the OpenRouter/Minimax proxy) so that I can compare review quality and cost against the OR preset.

## Context

`.github/workflows/claude-code-review.yml` lines 35–52 document a native-Anthropic auth convention:

> Native Anthropic (ANTHROPIC_BASE_URL empty): anthropic_api_key input carries ANTHROPIC_API_KEY (x-api-key header). ANTHROPIC_AUTH_TOKEN left empty.

Deleted the three `ANTHROPIC_*` repo vars (per [run 24478516759](https://github.com/hubertgajewski/orwellstat/actions/runs/24478516759), 2026-04-15) to exercise this path. The action step failed with 3 consecutive retries:

```
Attempt 1 of 3...
Attempt 1 failed: Unable to connect. Is the computer able to access the url?
(135s per attempt; total step time ~7 min)
##[error]Action failed with error: Unable to connect. Is the computer able to access the url?
```

`ANTHROPIC_API_KEY` secret exists (updated 2026-04-10) and `ANTHROPIC_BASE_URL` logged as empty (as the workflow's conditional intends). The SDK init options also showed `"claude_code_oauth_token": ""` — the v1 action may prefer OAuth over `anthropic_api_key` on the native path.

Every claude-code-review run in recent history (grep of last 20 runs) used OR+Minimax. Last explicit native-model commit was `27b54a4 "89 Pin claude-code-action model to claude-sonnet-4-6"`, predating the v1 action migration. So native may have never been exercised under v1.

## Acceptance Criteria

- **Given** `vars.ANTHROPIC_BASE_URL` is unset/empty (and `ANTHROPIC_API_KEY` secret is valid), **when** `claude-code-review.yml` fires on a PR, **then** the `anthropics/claude-code-action@v1` step authenticates successfully against `api.anthropic.com` and posts a review.
- **Given** the action step fails on native, **when** we read its log, **then** the actual URL it can't reach is visible (not hidden behind "full output hidden for security").
- **Given** the native path requires `CLAUDE_CODE_OAUTH_TOKEN` in addition to or instead of `anthropic_api_key`, **then** the workflow is updated to pass it, the secret is provisioned, and the auth convention comment block at lines 35–44 is updated to reflect the new contract.

## Implementation Hint

Debugging steps:

1. Add `show_full_output: true` to the action's `with:` block on `main` (one-off PR, own run will 401 on the workflow-validation sandbox — expected; merge anyway).
2. Flip the three `ANTHROPIC_*` vars empty on a post-merge test PR; observe the unmasked URL in the log.
3. If the URL is Anthropic's managed-agents OAuth endpoint, provision `CLAUDE_CODE_OAUTH_TOKEN` via `claude setup-token` and add it to `with:` alongside `anthropic_api_key`.
4. If the URL is `api.anthropic.com` directly, re-validate the `ANTHROPIC_API_KEY` secret (may be stale/rotated).
5. Remove `show_full_output: true` in a follow-up PR once the root cause is understood.

## Definition of Done

- [x] Native path is proven to work end-to-end: a real review is posted by `claude[bot]` on a PR where `vars.ANTHROPIC_BASE_URL` is empty. (PR #268 reviewer run on `feature/266`.)
- [x] Signature line of that review reads `_Reviewed by native Anthropic/claude-sonnet-4-6_` (or the configured native model slug). (Confirmed verbatim on PR #268.)
- [x] Workflow comment block at lines 35–44 reflects the final auth contract (what secrets are required). (`.github/workflows/claude-code-review.yml:40-49` — comments match the line-58 ternary; no contract change was required once the native path was proven working.)
- [x] `show_full_output: true` is not left in the production workflow. (Removed in commit `c3e8965 #262 Remove diagnostic show_full_output after native path confirmed working`.)
- [x] Repo vars are restored to OR+Minimax after the test. (PR #289 review signature `_Reviewed by proxy/@preset/minimax-minimax-m2-7-no-thinking_` confirms OR+Minimax is the active config.)

--- a/.github/workflows/claude-code-review.yml
+++ b/.github/workflows/claude-code-review.yml
@@ -56,6 +56,9 @@ jobs:
         with:
           anthropic_api_key: ${{ (vars.ANTHROPIC_BASE_URL != '' && secrets.OPENROUTER_API_KEY) || (vars.ANTHROPIC_BASE_URL == '' && secrets.ANTHROPIC_API_KEY) || '' }}
           track_progress: false
+          # Diagnostic — unmask the URL on connection failures so issue #262
+          # can root-cause the native-Anthropic auth path. Remove once resolved.
+          show_full_output: true
           prompt: |
             You are executing a code review of pull request #${{ github.event.pull_request.number }}
             in ${{ github.repository }}. This is an autonomous task: complete it using tool calls

